### PR TITLE
Feature/redux and del button

### PR DESCRIPTION
**Pull request by MikhailovIgor**
added redux, added deleting logic from store for the product

### CODE REVIEW

File: js/index.js [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
 
-const fetchProducts =  async () => {
-  const result = await fetch(URL);
-  return await result.json();
-}
+const store = createStore(productReducer, products);
 
-const render = data => data.forEach(item => items.insertAdjacentHTML('beforeend', card(item)));
+const render = data => {
+  items.querySelectorAll('.card').forEach(item => item.remove()) // items.innerHTML='';
```
Comment (sablevsky) [inline js/index.js:13]: Unnecessary comment

File: js/redux/actions.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import {DELETE_PRODUCT} from './types.js'
+
+export const deleteProduct = (id) => ({type: DELETE_PRODUCT, payload: id});
```
Comment (sablevsky) [inline js/redux/actions.js:3]: No newline at end of file

File: js/redux/productInitialState.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    image: "img/06.jpeg",
+    description: "Android 9.0 Pie, Qualcomm Snapdragon 439, 2/16GB, 13 MPX camera, Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+    price: 389,
+    discount: null,
+    labels: {
+      hot: true,
+      new: true
+    }
+  }
+]
```
Comment (sablevsky) [inline js/redux/productInitialState.js:74]: No newline at end of file

File: js/redux/productReducer.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+import {DELETE_PRODUCT} from "./types.js";
+
+export function productReducer(state= products, action) {
+  console.log(action.type);
+  switch (action.type) {
+    case DELETE_PRODUCT: return state.filter(item => item.id !== action.payload)
+    
+    default: return state;
+  }
+}
```
Comment (sablevsky) [inline js/redux/productReducer.js:11]: No newline at end of file

File: js/redux/store.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  };
+  
+  const subscribe = callback => {
+    subscribers.push(callback);
+  };
+  
+  return {
+    getState, dispatch, subscribe
+  };
+}
```
Comment (sablevsky) [inline js/redux/store.js:21]: No newline at end of file

File: js/redux/types.js [hunk 6]
```diff
@@ -0,0 +1,2 @@
+export const INIT = 'INIT';
+export const DELETE_PRODUCT = 'DELETE PRODUCT';
```
Comment (sablevsky) [inline js/redux/types.js:2]: No newline at end of file

File: css/card.css [hunk 7]
```diff
@@ -67,59 +67,20 @@
...
-
-
-
-
-
-
-    padding: 2px 5px 0;
+.card__delete:after {
+    content: 'X';
 }
```
Comment (sablevsky) [inline css/card.css:86]: No newline at end of file

**Review by sablevsky — CHANGES_REQUESTED** [review]
There are some issues

File: js/redux/store.js [hunk 8]
```diff
@@ -18,4 +18,4 @@ export function createStore(reducer, initialState) {
   return {
     getState, dispatch, subscribe
   };
-}
\ No newline at end of file
+}
```
Comment (sablevsky) [inline js/redux/store.js:21]: missed semicolon

File: js/redux/productReducer.js [hunk 9]
```diff
@@ -8,4 +8,4 @@ export function productReducer(state= products, action) {
     
     default: return state;
   }
-}
\ No newline at end of file
+}
```
Comment (sablevsky) [inline js/redux/productReducer.js:11]: Missed semicolon

**Review by sablevsky — CHANGES_REQUESTED** [review]

File: js/redux/store.js (hunk 8 above)
Comment (MikhailovIgor) [reply js/redux/store.js:21]: function declaration doesn't need a semicolon at the end (after curly brace)

File: js/redux/productReducer.js (hunk 9 above)
Comment (MikhailovIgor) [reply js/redux/productReducer.js:11]: function declaration doesn't need a semicolon at the end (after curly brace)

**Pull request merged**